src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.InvalidUserDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(null);
+        });
+        assertEquals("Користувач не може бути null", exception.getMessage());
+    }
+
+    @Test
+    void register_userWithExistingLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("testuser");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("testuser");
+        user2.setPassword("newpassword");
+        user2.setAge(22);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user2);
+        });
+        assertEquals("Користувач з таким логіном вже існує", exception.getMessage());
+    }
```

Comment:
According to the checklist, you should not use the `register` method to add users for testing. Instead, you should use low-level methods like `Storage.people.add()`. This is because we cannot guarantee the correctness of the `register` method yet.